core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -53,14 +53,22 @@ import org.apache.spark.util.{NextIterator, SerializableConfiguration, ShutdownH
 /**
  * A Spark split class that wraps around a Hadoop InputSplit.
  */
-private[spark] class HadoopPartition(rddId: Int, idx: Int, s: InputSplit)
+private[spark] class HadoopPartition(rddId: Int, override val index: Int, s: InputSplit)
   extends Partition {
 
   val inputSplit = new SerializableWritable[InputSplit](s)
 
-  override def hashCode(): Int = 41 * (41 + rddId) + idx
+  override def hashCode(): Int = 41 * (41 + rddId) + index
 
-  override val index: Int = idx
+  def canEqual(other: Any): Boolean = other.isInstanceOf[HadoopPartition]
+
+  override def equals(other: Any): Boolean = other match {
+    case that: HadoopPartition =>
+      super.equals(that) &&
+        (that canEqual this) &&
+        index == that.index
```

Comment:
Right. So I feel like there is many cases where we might have this ambiguity and this might be out of the scope for this Jira/PR which is to enable the style check.

So would it be ok if we just do `override def equals(other: Any): Boolean = super.equals(other)` where ever there is an ambiguity in order to enable the stylecheck and avoid further implementation of orphan hashCode/equals?